### DISCUSSION TITLE
No more detections

### DISCUSSION THREAD

**Discussion by blogdemoi** [mcguirepr89/BirdNET-Pi#614] [Q&A]
Hello,

Two days ago I had tout unplug my Pi without shutting down Birdnet-pi properly. Since then there are no more detections although Birdnet-Pi seems to work as usual, web interface is OK, services are up (I restarted them to be sure), live audio is working...

What could be wrong?
Any help welcome.

Thank you.

**Comment by ehpersonal38** [comment]
I'm going to see logs of some sort to even have a clue of what's happening here. Can you get a screenshot or copy and paste your "View Logs" output?

**Reply by blogdemoi** [reply]
I am sorry @ehpersonal38 for the late reply, I missed your reply. I left the Pi unattended for 2 days and after a restart it seems that the .db database was corrupted as it had only the last 2 days detections. I supposed unplugging from power, several times, my Birdnet-Pi didn't make any good to it... Fortunately I make backups of the .db file every week so I tried to import my last backup (I had to delete the current .db used in Birdnet-Pi first) and...success.

Then I also re-imported the files in the Extracted folder (I also made a backup). It worked fine. However I noticed my audio recordings marked as not to be purged were not marked not to be purged anymore. In which file are they marked not to be purged? Is it in the birdnet.conf? (I didn't restore that one).

So now I am thinking of automatizing backups the following way:
A cron job that will launch every night a rsync job to save the .db file and the new recordings in the Extracted folder, to my local server at home (I never tried rsync before).

What do you think? Do I have to stop any services before launching rsync to backup the .db file?
Also do you know any good tool to merge .db files?

Thank you.

**Reply by ehpersonal38** [reply]
No worries!

> In which file are they marked not to be purged?

I believe the file you're looking for is `~/BirdNET-Pi/scripts/disk_check_exclude.txt`.

> So now I am thinking of automatizing backups the following way:

Yes I have rsync set up in a similar way, One note though is that you will have to copy birds.db to a temporary file, and then sync that temporary file, rather than straight up syncing birds.db. This is a sample script that works for me:

```
#!/bin/bash

cp /home/pi/BirdNET-Pi/scripts/birds.db /home/pi/BirdNET-Pi/scripts/birds_tmp.db
/usr/bin/rclone copy --update --verbose --transfers 30 --checkers 8 --contimeout 60s --timeout 300s --retries 3 --low-level-retries 10 --min-size 10M --stats 1s "/home/pi/BirdNET-Pi/scripts/birds_tmp.db" "google-drive:PiBackups"
```

> Also do you know any good tool to merge .db files?

Honestly no clue, sorry.

**Reply by blogdemoi** [reply]
Thank you.

**Comment by blogdemoi** [comment]
I am wondering if the "No detections for today" sentence is at the correct place, on the landing page of my Birdnet-Pi... Shouldn't it be in the center and not on the left ?

![image](https://i.imgur.com/t5pKxAk.png)

**Reply by ehpersonal38** [reply]
Yeah I suppose it could use a center tag, that's pretty minor though. I'll see if I can get around to it soon
